src/Microsoft.AspNetCore.Mvc.Core/Routing/UrlHelper.cs

```diff
@@ -177,29 +230,29 @@ protected virtual VirtualPathData GetVirtualPathData(string routeName, RouteValu
         }
 
         // Internal for unit testing.
-        internal void AppendPathAndFragment(StringBuilder builder, VirtualPathData pathData, string fragment)
+        internal void AppendPathAndFragment(StringBuilder builder, string virtualPath, string fragment)
```

Comment:
With separate implementations move shared code like this to an internal helper class